### DISCUSSION TITLE
AsDiscreted transform: what does thresholding do after argmax ?

### DISCUSSION THREAD

**Discussion by franciskasara** [Project-MONAI/tutorials#1612] [Q&A]
In `AsDiscreted(keys="pred", argmax=True,threshold=0.5)` the thresholding step is after the argmax is taken. Does threshold after argmax does anything?

**Comment by KumoLiu** [comment]
Hi @franciskasara, thresholding is used to obtain discrete values.
Thanks.

**Reply by franciskasara** [reply]
But after argmax, aren't all values discrete? I know what thresholding does in general, I just don't understand its role after argmax.
Could you give me an example, where the result of argmax without threshold going to be different than the result with argmax and then threshold?

**Reply by KumoLiu** [reply]
So I don't think you need to add a threshold after argmax, right? They are all optional args.

**Reply by franciskasara** [reply]
What confuses me is that in the MONAI documentation this is one of the examples:  `transform = AsDiscrete(argmax=True, to_onehot=2, threshold=0.5)` . And that suggest to me that there are cases when you need both the argmax and the threshold parameters.
